lightning/src/blinded_path/message.rs

```diff
@@ -402,6 +403,28 @@ pub enum AsyncPaymentsContext {
 		/// containing the expected [`PaymentId`].
 		hmac: Hmac<Sha256>,
 	},
+	/// Context contained within the [`BlindedMessagePath`]s we put in static invoices, provided back
+	/// to us in corresponding [`HeldHtlcAvailable`] messages.
+	///
+	/// [`HeldHtlcAvailable`]: crate::onion_message::async_payments::HeldHtlcAvailable
+	InboundPayment {
+		/// The ID of the [`Offer`] that this [`BlindedMessagePath`]'s static invoice corresponds to.
+		/// Useful to authenticate that this blinded path was created by us for asynchronously paying
+		/// one of our offers.
+		///
+		/// [`Offer`]: crate::offers::offer::Offer
+		offer_id: OfferId,
```

Comment:
This is basically just used as an input into the HMAC field 10 lines below, so it helps authenticate this context. But maybe the nonce is sufficient as a unique input? I was following the pattern of `hmac_for_payment_id` used for regular bolt 12 invoices but using an `offer_id` instead.

Reply:
Yeah, I didn't really have a sense of whether nonce + hmac was sufficient since we haven't used only those things previously, but since it is I went ahead and removed it. Will track expiring these paths on the async payments issue.

Reply:
To be clear, in this scenario the node trying to ask if we're online has our node id. Another way for nodes to trivially ask if we're online is just by attempting to connect to us as a peer -- but presumably in this case the attacker is prevented from doing it that way by not knowing our socket address?

Reply:
Ok I started doing this but tbh I'd prefer to save it for follow-up if it's all the same to you. 

> No, I'm thinking of a mobile node that's only connect to an LSP and for which we only have a blinded path to.

Ok so for context I think that means the attacker would be reusing the `blinded_node_id`s to construct their path. What I'm not following is that in order to blind the TLVs of their choosing, I thought they would need the shared secret corresponding to the node's unblinded pubkey and the blinding point?

Reply:
> Arguably it would be reasonable to reply to _every_ inbound `HeldHtlcAvailable` message by just asking that the HTLC be released. It would imply someone can trivially ask if/when you're online which isn't ideal, so we should have _something_ here (a nonce/hmac pair seems to suffice), but ultimately we may get a bunch of these for any given offer, so its not like we can track them and remove offers after they're paid. We might want a timeout here so that we can stop replying to old ones that have expired, I guess?

IIUC, this context is in the static invoice's message paths. So couldn't an adversary simply request an invoice, get back the static invoice, and save the path to use whenever it wants? We could have it expire with the static invoice, but there doesn't seem to be anything that would stop the adversary from doing the same thing with a new static invoice. Am I missing something?

Reply:
You're not missing anything IIUC. The only thing that limits the attack you're describing is also expiring the offer/its paths.

Reply:
But isn't this a *message* path from the static invoice? Would we need to add an expiration to this new context in order to expire it? For payment paths we'd use the payment secret, IIUC.

Reply:
Discussed offline; it is the case that if a recipient publishes an offer that never expires, then anyone can pretty much always check if they're online under this protocol as long as the offer's message paths remain valid. Probably worth noting in documentation in the future.

Reply:
Chatted offline. Just was trying to point out that if you _always_ want to be paid by _anyone_, then anyone can check if you are online because presumably you will always have a fresh offer with unexpired paths published.